From 8b3fb89340721b52395df32bf9ed83d6cbafc59f Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Sat, 12 Oct 2019 10:18:41 +0800
Subject: [PATCH] input: sensors: gyro: fix clang warning

drivers/input/sensors/gyro/l3g4200d.c:175:21: warning: address of array 'pdata->orientation' will always evaluate to 'true' [-Wpointer-bool-conversion]
        if(pdata && pdata->orientation)
                 ~~ ~~~~~~~^~~~~~~~~~~

drivers/input/sensors/gyro/l3g20d.c:175:21: warning: address of array 'pdata->orientation' will always evaluate to 'true' [-Wpointer-bool-conversion]
        if(pdata && pdata->orientation)
                 ~~ ~~~~~~~^~~~~~~~~~~

drivers/input/sensors/gyro/ewtsa.c:386:21: warning: address of array 'pdata->orientation' will always evaluate to 'true' [-Wpointer-bool-conversion]
        if(pdata && pdata->orientation)
                 ~~ ~~~~~~~^~~~~~~~~~~

Change-Id: Ic4f9f33ac1654711434a9ab383da02bfade5ab13
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/input/sensors/gyro/ewtsa.c    | 2 +-
 drivers/input/sensors/gyro/l3g20d.c   | 2 +-
 drivers/input/sensors/gyro/l3g4200d.c | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/input/sensors/gyro/ewtsa.c b/drivers/input/sensors/gyro/ewtsa.c
index 4756ed24c0a7..0a8534eb24a1 100644
--- a/drivers/input/sensors/gyro/ewtsa.c
+++ b/drivers/input/sensors/gyro/ewtsa.c
@@ -383,7 +383,7 @@ static int sensor_report_value(struct i2c_client *client)
 	z = (short) (((buffer[4]) << 8) | buffer[5]);
 
 	//printk("%s: x=%d  y=%d z=%d \n",__func__, x,y,z);
-	if(pdata && pdata->orientation)
+	if (pdata)
 	{
 		axis.x = (pdata->orientation[0])*x + (pdata->orientation[1])*y + (pdata->orientation[2])*z;
 		axis.y = (pdata->orientation[3])*x + (pdata->orientation[4])*y + (pdata->orientation[5])*z;
diff --git a/drivers/input/sensors/gyro/l3g20d.c b/drivers/input/sensors/gyro/l3g20d.c
index 300acd3f28ca..ffa3b71f1938 100644
--- a/drivers/input/sensors/gyro/l3g20d.c
+++ b/drivers/input/sensors/gyro/l3g20d.c
@@ -172,7 +172,7 @@ static int sensor_report_value(struct i2c_client *client)
 	z = (short) (((buffer[5]) << 8) | buffer[4]);
 
 	DBG("%s: x=%d  y=%d z=%d \n",__func__, x,y,z);
-	if(pdata && pdata->orientation)
+	if (pdata)
 	{
 		axis.x = (pdata->orientation[0])*x + (pdata->orientation[1])*y + (pdata->orientation[2])*z;
 		axis.y = (pdata->orientation[3])*x + (pdata->orientation[4])*y + (pdata->orientation[5])*z;
diff --git a/drivers/input/sensors/gyro/l3g4200d.c b/drivers/input/sensors/gyro/l3g4200d.c
index 508c907b1905..4f236acd7118 100644
--- a/drivers/input/sensors/gyro/l3g4200d.c
+++ b/drivers/input/sensors/gyro/l3g4200d.c
@@ -172,7 +172,7 @@ static int sensor_report_value(struct i2c_client *client)
 	z = (short) (((buffer[5]) << 8) | buffer[4]);
 
 	DBG("%s: x=%d  y=%d z=%d \n",__func__, x,y,z);
-	if(pdata && pdata->orientation)
+	if (pdata)
 	{
 		axis.x = (pdata->orientation[0])*x + (pdata->orientation[1])*y + (pdata->orientation[2])*z;
 		axis.y = (pdata->orientation[3])*x + (pdata->orientation[4])*y + (pdata->orientation[5])*z;
-- 
2.35.3

